src/Validator/Constraints/ImageUploadDimensionValidator.php

```diff
@@ -28,11 +28,11 @@ final class ImageUploadDimensionValidator extends ConstraintValidator
     private $imagineAdapter;
 
     /**
-     * @var ImageProvider
+     * @var ImageProviderInterface
      */
     private $imageProvider;
 
-    public function __construct(ImagineInterface $imagineAdapter, ImageProvider $imageProvider)
+    public function __construct(ImagineInterface $imagineAdapter, ImageProviderInterface $imageProvider)
```

Comment:
Not sure but this change looks like BC break to me.

Reply:
I can only think of one situation, namely when someone extends the validator and its constructor.

Reply:
Why wouldn't it be BC @jordisala1991 ?

Reply:
I could also remove the explicit parameter type and check it at runtime.

Reply:
If it is BC, fine by me :)